components/snackbar.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+  <v-snackbar v-model="open" :timeout="3000" :color="variant">
+    <span :style="{ color: textColor }">{{ message }}</span>
+    <v-btn :style="{ color: textColor }" text @click="open = false"
+      >Close</v-btn
+    >
+  </v-snackbar>
+</template>
+
+<script lang="ts">
+export default {
+  data() {
+    return {
+      open: false,
+      variant: '',
+      message: '',
+    }
+  },
+  computed: {
+    textColor: function () {
+      const variantToTextColorMap = {
+        success: '#082218',
+        warning: '#2A150D',
+        error: '#FFF8F8',
+        info: '#050709',
+      }
+      return variantToTextColorMap[this.variant]
+    },
+  },
+  created: function () {
+    this.$store.subscribe((mutation, state) => {
+      if (mutation.type === 'snackbar/showSnackbar') {
+        const message = state.snackbar.message
+        const variant = isAcceptedVariant(state.snackbar.variant)
+          ? state.snackbar.variant
+          : 'info'
+
+        if (message) {
+          this.open = true
+          this.message = message
+          this.variant = variant
+        }
+      }
+    })
+  },
+}
+
+const isAcceptedVariant = (variant) => {
+  return (
+    variant === 'info' ||
+    variant === 'success' ||
+    variant === 'warning' ||
+    variant === 'error'
+  )
+}
```

Comment:
Any reason this function is not inside the `methods` object?

Reply:
Not in particular, no. I didn't really know where you put these kind of helper functions in Vue. I assumed that you only used the methods for functions that the template needs to have access to, but that might be wrong.
As Oscar puts it: "I'm a React grill" I have no clue what I'm doing in Vue 😄

I can make another PR to fix that if we want.

Reply:
Ah yes that would be wonderful, thanks. No worries, I was also a bit too quick to merge the PR, I just skimmed it briefly and merged it. Should have reviewed everything in detail first.

Reply:
> Should have reviewed everything in detail first.

Hehehe, that might have been a good idea yeah 😄